From 23e2deb5f5f4161c8268446936e47d423905060c Mon Sep 17 00:00:00 2001
From: CruX <crux@project-insanity.org>
Date: Tue, 21 Oct 2014 08:43:19 +0200
Subject: [PATCH] fix fec initialization problems

This fixes fec initialization problems observed on the wandboard quad.
Without this fix, the boot process did not work reliably.
Fix proposed by rmk, thanks.
---
 drivers/net/ethernet/freescale/fec_main.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/freescale/fec_main.c b/drivers/net/ethernet/freescale/fec_main.c
index eba6447..a59a125 100644
--- a/drivers/net/ethernet/freescale/fec_main.c
+++ b/drivers/net/ethernet/freescale/fec_main.c
@@ -2704,6 +2704,8 @@ failed_irq:
 failed_init:
 	if (fep->reg_phy)
 		regulator_disable(fep->reg_phy);
+	if (fep->ptp_clock)
+		ptp_clock_unregister(fep->ptp_clock);
 failed_regulator:
 	fec_enet_clk_enable(ndev, false);
 failed_clk:
-- 
1.7.9.5

